documentation/en/mpool.md

```diff
@@ -0,0 +1,155 @@
+lotus mpool find [--from <address>] [--to <address>] [--method <int>]
+lotus mpool config [<configuration>]
+```
+
+### lotus mpool pending
+Prints the pending messages in the mpool, as returned by the `MpoolPending` API call.
+If `--local` is specified, it only prints pending messages for addresses in the local wallet.
+
+### lotus mpool sub
+Subscribes to mpool changes using the `MpoolSub` API call and prints the stream of mpool
+updates.
+
+### lotus mpool stat
+Prints various mpool statistics.
+If `--local` is specified then only prints statistics for addresses in the local wallet.
+
+### lotus mpool replace
+Replaces a message in the mpool.
+
+### lotus mpool find
+Searches for messages in the mpool.
+
+### lotus mpool config
+Gets or sets the current mpool configuration.
+
+## Configuration
+
+The mpool a few parameters that can be configured by the user, either through the API
+or the command line interface.
+
+The config struct is defined as follows:
+```
+type MpoolConfig struct {
+	PriorityAddrs          []address.Address
+	SizeLimitHigh          int
+	SizeLimitLow           int
+	ReplaceByFeeRatio      float64
+	PruneCooldown          time.Duration
+	GasLimitOverestimation float64
+}
+
+```
+
+The meaning of these fields is as follows:
+- `PriorityAddrs` -- these are the addresses of actors whose pending messages should always
+  be included in a block during message selection, regardless of profitability.
+  Miners should configure their own worker addresses so that they include their own messages
+  when they produce a new block.
+- `SizeLimitHigh` -- this is the maximum number of pending messages before triggering a
+  prune in the message pool. Note that messages from priority addresses are never pruned.
+- `SizeLimitLow` -- this is the number of pending messages that should be kept after a prune.
+- `ReplaceByFeeRatio` -- this is the gas fee ratio for replacing messages in the mpool.
+  Whenever a message is replaced, the `GasPremium` must be increased by this ratio.
+- `PruneCooldown` -- this is the period of time to wait before triggering a new prune.
+- `GasLimitOverestimation` -- this is a parameter that controls the gas limit overestimation for new messages.
+
+
+## Message Selection
+
+A few words regarding message selection are pertinent. In message
+selection, a miner selects a set of messages from the pending messages
+for inclusion to the next block in order to maximize its reward. The
+problem however is NP-hard (it's an instance of knapsack packing),
+further confounded by the fact that miners don't communicate their
+tickets to each other. So at best we can approximate the optimal
+selection in a reasonable amount of time.
+
+The mpool employs a sophisticated algorithm for selecting messages for
+inclusion, given the ticket quality of a miner. The ticket quality
+reflects the probability that a block will execute in the
+tipset. Given the ticket quality the algorithm computes the
```

Comment:
The wording is mildly confusing here. All blocks in a tipset execute, it's just the first one to include a given message which gets the reward

Reply:
Yeah, the word should probably be message.